lib/Interpreter/IncrementalParser.cpp

```diff
@@ -206,46 +226,65 @@ namespace cling {
       = m_CI->getInvocation().getPreprocessorOpts().ImplicitPCHInclude;
     if (!PCHFileName.empty()) {
       Transaction* CurT = beginTransaction(CO);
+      DiagnosticErrorTrap Trap(m_CI->getSema().getDiagnostics());
       m_CI->createPCHExternalASTSource(PCHFileName,
                                        true /*DisablePCHValidation*/,
                                        true /*AllowPCHWithCompilerErrors*/,
                                        0 /*DeserializationListener*/,
                                        true /*OwnsDeserializationListener*/);
+
+      // TODO: Better diagnosis of whether the error is safe to ignore.
```

Comment:
Could you transform that into FIXME, we should be using FIXME all over for easier grepping. Also, it seems this check is enough. If the error recovery cannot unload a transaction it should set this bit to bail out.